uni/lib/view/home/widgets/schedule_card.dart

```diff
@@ -51,53 +55,55 @@ class ScheduleCard extends GenericCard {
     );
   }
 
-  Widget generateSchedule(
-    UnmodifiableListView<Lecture> lectures,
-    BuildContext context,
-  ) {
-    final lectureList = List<Lecture>.of(lectures);
-    return Column(
-      mainAxisSize: MainAxisSize.min,
-      children: getScheduleRows(context, lectureList),
-    );
-  }
-
   List<Widget> getScheduleRows(BuildContext context, List<Lecture> lectures) {
     final rows = <Widget>[];
 
-    final now = DateTime.now();
-    var added = 0; // Lectures added to widget
-    var lastAddedLectureDate = DateTime.now(); // Day of last added lecture
-
-    for (var i = 0; added < 2 && i < lectures.length; i++) {
-      if (now.compareTo(lectures[i].endTime) < 0) {
-        if (lastAddedLectureDate.weekday != lectures[i].startTime.weekday &&
-            lastAddedLectureDate.compareTo(lectures[i].startTime) <= 0) {
-          rows.add(
-            DateRectangle(
-              date:
-                  Provider.of<LocaleNotifier>(context).getWeekdaysWithLocale()[
-                      (lectures[i].startTime.weekday - 1) % 7],
-            ),
-          );
-        }
-
-        rows.add(createRowFromLecture(context, lectures[i]));
-        lastAddedLectureDate = lectures[i].startTime;
-        added++;
+    final lecturesByDay = lectures
+        .groupListsBy(
+          (lecture) => lecture.startTime.weekday,
+        )
+        .entries
+        .toList()
+        .sortedBy<num>((element) {
+      // Sort by day of the week, but next days come first
+      final dayDiff = element.key - DateTime.now().weekday;
+      return dayDiff >= 0 ? dayDiff - 7 : dayDiff;
+    }).toList();
+
+    for (final dayLectures
+        in lecturesByDay.sublist(0, min(2, lecturesByDay.length))) {
+      final day = dayLectures.key;
+      final lectures = dayLectures.value
+          .where(
+            (element) =>
+                // Hide finished lectures from today
+                element.startTime.weekday != DateTime.now().weekday ||
```

Comment:
Is `startTime` based on the current date? Or is it based on a generic date like 1-1-1 
in `DateTime(1, 1, 1, hour, minute, second)`? Sorry, I am not very familiar with how uni stores datetimes...

Reply:
Start time is the actual time of the class in that week, including the day and month. Sigarra returns the complete date for each week

Reply:
Awesome!